### PR TITLE
RabbitMQ native extension

### PR DESCRIPTION
**Pull request by Jeansen**
Promote RabbitMQ to natvive, see #1531

### CODE REVIEW

File: tooling/scripts/test-categories.yaml [hunk 1]
```diff
@@ -75,6 +75,7 @@ memoryhog-xml:
   - rest-binding-mode-xml
   - xml
 messaging:
+  - rabbitmq
```
[the remark this answers is not in the document]
Comment (Jeansen) [reply tooling/scripts/test-categories.yaml]: Ok. I've corrected hat one.

**Comment by Jeansen** [comment]
After rebasing, local test still passes ;-)

**Comment by ppalaga** [comment]
The CI says you should build locally and amend the reformatted Java file and the generated AsciiDoc files.

**Comment by Jeansen** [comment]
> The CI says you should build locally and amend the reformatted Java file and the generated AsciiDoc files.

When I run ` mvn clean install -DskipTests -Pformat` from within my integration test, I do not get any errors or additional files to add.

**Comment by ppalaga** [comment]
> But when run from the main folder, I get the following:

Yes, that's what I meant. Sorry that I have not made it clear straight away.

> ```
> Changes not staged for commit:
>   (use "git add <file>..." to update what will be committed)
>   (use "git restore <file>..." to discard changes in working directory)
> 	modified:   docs/modules/ROOT/pages/reference/components.adoc
> 	modified:   docs/modules/ROOT/pages/reference/extensions/rabbitmq.adoc
> 	modified:   docs/modules/ROOT/pages/reference/index.adoc
> 	modified:   extensions/rabbitmq/deployment/src/main/java/org/apache/camel/quarkus/component/rabbitmq/deployment/RabbitmqProcessor.java
> 	modified:   extensions/rabbitmq/runtime/src/main/resources/META-INF/quarkus-extension.yaml
> 	modified:   integration-tests/rabbitmq/pom.xml

All of the above definitely need to be amended. 

But the changes in `extensions-jvm/apns` seem strange. Please ignore them.

> Untracked files:
>   (use "git add <file>..." to include in what will be committed)
> 	extensions-jvm/apns/deployment/src/main/java/org/apache/camel/quarkus/component/rabbitmq/
> 	extensions-jvm/apns/integration-test/src/main/java/org/apache/camel/quarkus/component/rabbitmq/
> 	extensions-jvm/apns/integration-test/src/test/java/org/apache/camel/quarkus/component/apns/it/RabbitmqTest.java

**Comment by Jeansen** [comment]
> > But when run from the main folder, I get the following:
> 
> Yes, that's what I meant. Sorry that I have not made it clear straight away.

Not a problem. I'ts a big project. I've amended the changes.

**Comment by ppalaga** [comment]
Great work @Jeansen , thanks a lot!

**Pull request merged**